processor/src/main/java/org/mapstruct/ap/internal/model/IterableMappingMethod.java

```diff
@@ -202,6 +203,10 @@ private Assignment forgeMapping(SourceRHS sourceRHS, Type sourceType, Type targe
             return assignment;
         }
 
+        private String stubPropertyName(String fullyQualifiedName) {
+            return Strings.decapitalize( fullyQualifiedName.substring( fullyQualifiedName.lastIndexOf( '.' ) + 1 ) );
```

Comment:
Doing this here for deriving a property name feels wrong... is it only used for adding a postfix after `[].` / `{}.` for iterable/maps?